File: 287-FindTheDuplicateNumber/287-FindTheDuplicateNumber.py
# Last updated: 4/21/2026, 11:25:41 PM
from collections import Counter
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        slow = nums[0]
        fast = nums[0]
        while True:
            slow = nums[slow]
            fast = nums[nums[fast]]
            if slow == fast:
                break
            
        slow = nums[0]
        while slow != fast:
            slow = nums[slow]
            fast = nums[fast]
        return fast

File: 287-FindtheDuplicateNumber/287-FindtheDuplicateNumber.java
// Last updated: 8/19/2026, 2:54:05 PM
1class Solution {
2    public int findDuplicate(int[] nums) {
3       int slow = 0;
4       int fast = 0;
5       while(fast < nums.length){
6         slow = nums[slow];
7         fast = nums[nums[fast]];
8         if(slow == fast){
9            break;
10        }
11       }
12        int slow2 = 0;
13        while (slow2 != fast){
14            slow2 = nums[slow2];
15            fast= nums[fast];
16        }
17        return slow2;
18       
19    }       
20}